src/components/Form/Form.js

```diff
@@ -169,11 +173,17 @@ export function Form({ children, config }) {
       fieldComponent.labelCopy = fieldConfig.labelCopy
     }
 
+    // We need to be able to to signify that the form is in a valid state
+    // regardless of whether an optional field is "filled out" or not.
+    if (fieldConfig.optional) {
+      fieldComponent.optional = fieldConfig.optional
+    }
+
     return fieldConfig.component(
       fieldComponent,
 
       // For things like ButtonGroupField, which may have options supplied.
-      fieldConfig.options || null
+      fieldConfig.options || undefined
```

Comment:
very minor nit but i would like it if we wrote somewhere that this repo uses undefined, not null, for null-like values or whatever you would call them. with the validateExists logic it seems like this is a pattern and it may squelch future arguments if we write it down somewhere(?)

Reply:
this doesn't really need to be in this pr though

Reply:
Good point. Let me think about this. There's actually a ton of documentation missing/needed but just haven't got to it. But I agree this might be nice to codify somehow 👍